### PR TITLE
Handle shp2wpkh type correctly in createDescriptor

### PR DESCRIPTION
**Pull request by shobitb**
With the way the switch/case block is written in createDescriptor, the shp2wpkh and p2shp2wpkh cases result in a descriptor without a closing parenthesis e.g. `sh(wpkh(...)`. This change is a refactoring of the switch case block to enable constructing descriptor for all cases correctly, and also improve maintainability and readability.

### CODE REVIEW

**Comment by BitcoinZavior** [comment]
@shobitb Thanks for the PR. please rebase with master and I will merge.

**Comment by shobitb** [comment]
Hi @BitcoinZavior I've resolved the conflicts.  I did it via github (browser) so I think it created a new merge commit (not rebase).  Is that okay?

**Comment by BitcoinZavior** [comment]
> 

Yes that should do. I am just waiting for @rsbkme to try main branch before merge anything new.

**Review by BitcoinZavior — APPROVED** [review]

**Pull request merged**
